pkg/snapshotterutil/sociutil.go

```diff
@@ -0,0 +1,161 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package snapshotterutil
+
+import (
+	"bufio"
+	"os"
+	"os/exec"
+	"strconv"
+
+	"github.com/containerd/nerdctl/pkg/api/types"
+	"github.com/sirupsen/logrus"
+)
+
+// CreateSoci creates a SOCI index(`rawRef`)
+func CreateSoci(rawRef string, gOpts types.GlobalCommandOptions, allPlatform bool, platforms []string, sOpts types.SociOptions) error {
+	sociExecutable, err := exec.LookPath("soci")
+	if err != nil {
+		logrus.WithError(err).Error("soci executable not found in path $PATH")
+		logrus.Info("you might consider installing soci from: https://github.com/awslabs/soci-snapshotter/blob/main/docs/install.md")
+		return err
+	}
+
+	sociCmd := exec.Command(sociExecutable)
+	sociCmd.Env = os.Environ()
+
+	// #region for global flags.
+	if gOpts.Address != "" {
+		sociCmd.Args = append(sociCmd.Args, "--address", gOpts.Address)
+	}
+	if gOpts.Namespace != "" {
+		sociCmd.Args = append(sociCmd.Args, "--namespace", gOpts.Namespace)
+	}
+	// #endregion
+
+	// Global flags have to be put before subcommand before soci upgrades to urfave v3.
+	// https://github.com/urfave/cli/issues/1113
+	sociCmd.Args = append(sociCmd.Args, "create")
+
+	if allPlatform {
+		sociCmd.Args = append(sociCmd.Args, "--all-platforms", strconv.FormatBool(allPlatform))
+	}
+	if len(platforms) > 0 {
+		sociCmd.Args = append(sociCmd.Args, "--platform")
+		sociCmd.Args = append(sociCmd.Args, platforms...)
+	}
+	if gOpts.Namespace != "" {
+		sociCmd.Args = append(sociCmd.Args, "--namespace", gOpts.Namespace)
+	}
+	if sOpts.SpanSize != -1 {
+		sociCmd.Args = append(sociCmd.Args, "--span-size", strconv.FormatInt(sOpts.SpanSize, 10))
```

Comment:
Looks like values are in bytes in soci cli. See https://github.com/awslabs/soci-snapshotter/blob/f556b45e8e241ff243776e15352a6b7bd38f2161/soci/soci_index.go#L57-L58
```go
defaultSpanSize     = int64(1 << 22) // 4MiB
defaultMinLayerSize = 10 << 20       // 10MiB
```

Maybe room for improved UX to specify value with `mib`, `m`, `kib`,  `k` suffix and do the conversion?

Reply:
Yeah. Maybe this can contribute to soci CLI later.